yogstation/code/game/gamemodes/gangs/gangs.dm

```diff
@@ -0,0 +1,59 @@
+//gang.dm
+//Gang War Game Mode
+GLOBAL_LIST_INIT(possible_gangs, subtypesof(/datum/team/gang))
+GLOBAL_LIST_EMPTY(gangs)
+/datum/game_mode/gang
+	name = "gang war"
+	config_tag = "gang"
+	antag_flag = ROLE_GANG
+	restricted_jobs = list("Security Officer", "Warden", "Detective", "AI", "Cyborg","Captain", "Head of Personnel", "Head of Security")
+	required_players = 24
+	required_enemies = 1
+	recommended_enemies = 2
+	enemy_minimum_age = 14
+
+	announce_span = "danger"
+	announce_text = "A violent turf war has erupted on the station!\n\
+	<span class='danger'>Gangsters</span>: Take over the station with a dominator.\n\
+	<span class='notice'>Crew</span>: Prevent the gangs from expanding and initiating takeover."
+
+	var/list/datum/mind/gangboss_candidates = list()
+
+/datum/game_mode/gang/pre_setup()
+	if(CONFIG_GET(flag/protect_roles_from_antagonist))
+		restricted_jobs += protected_jobs
+
+	if(CONFIG_GET(flag/protect_assistant_from_antagonist))
+		restricted_jobs += "Assistant"
+
+	//Spawn more bosses depending on server population
+	var/gangs_to_create = 2
+	if(prob(num_players()) && num_players() > 1.5*required_players)
+		gangs_to_create++
+	if(prob(num_players()) && num_players() > 2*required_players)
+		gangs_to_create++
+	gangs_to_create = min(gangs_to_create, GLOB.possible_gangs.len)
+
+	for(var/i in 1 to gangs_to_create)
+		if(!antag_candidates.len)
+			break
+
+		//Now assign a boss for the gang
+		var/datum/mind/boss = pick_n_take(antag_candidates)
+		antag_candidates -= boss
+		gangboss_candidates += boss
+		boss.restricted_roles = restricted_jobs
+
+	if(gangboss_candidates.len < 1) //Need at least one gangs
+		return
+
+	return TRUE
+
+/datum/game_mode/gang/post_setup()
+	set waitfor = FALSE
```

Comment:
no other gamemode does this, why does this one?

Reply:
no clue